src/platform/intel/cavs/platform.c

```diff
@@ -388,11 +398,17 @@ int platform_init(struct sof *sof)
 #elif CONFIG_CANNONLAKE
 
 	/* initialize PM for boot */
+
+	/* request configured ring oscillator and wait for status ready */
+	shim_write(SHIM_CLKCTL, shim_read(SHIM_CLKCTL) | CNL_DEFAULT_RO);
+	while (!(shim_read(SHIM_CLKSTS) & CNL_DEFAULT_RO))
+		idelay(16);
+
 	shim_write(SHIM_CLKCTL,
-		   SHIM_CLKCTL_RHROSCC | /* Request High Performance RING Osc */
-		   SHIM_CLKCTL_OCS_HP_RING | /* Select HP RING Oscillator Clk
-					      * for memory
-					      */
+		   CNL_DEFAULT_RO | /* Request configured RING Osc */
```

Comment:
By default do you really mean default or *initial*? I'm going to add switching 120/400 clocks before entering/leaving waiti and if  that's really default then it should be kept f.e. 120 unless someone explicitly requests switching to higher clock.

Reply:
"Default" and "initial" terms are both used for the initial platform settings. Not sure which one is better since once the platform initialization is complete, any override with "initial" value should be interpreted as the "default". I am really not sure when you'd like to set sth back to the "initial" vs. to the "default" one at run-time?